### ISSUE TITLE
`invalid-length-returned` false positive when analysing "protocols"

### ISSUE THREAD

**Issue by ffissore** [PyCQA/pylint#4670]
### Steps to reproduce

Given a file `test.py`:
```python
from typing_extensions import Protocol


class TestProtocol(Protocol):

    def __len__(self) -> int:
        ...
```

### Current behavior

Running `poetry run pylint src/test.py` results in

```
************* Module test
src/test.py:6:4: E0303: __len__ does not return non-negative integer (invalid-length-returned)
```

### Expected behavior

Pylint should not raise warnings on protocol/empty methods

### pylint --version output

Tested with both latest 3x

```
pylint 3.0.0-a4
astroid 2.6.2
Python 3.7.9 (default, Sep 21 2020, 16:33:39) 
[GCC 9.3.0]
```
and latest 2.9

```
pylint 2.9.3
astroid 2.6.2
Python 3.7.9 (default, Sep 21 2020, 16:33:39) 
[GCC 9.3.0]
```

Additional dependencies:
```
typing-extensions==3.10.0.0
```

**Comment by ffissore** [comment]
This false positive didn't occur with pylint 2.8.3

**Comment by DanielNoord** [comment]
This has been fixed in a recent release 😄. Closing this!

**Comment by ffissore** [comment]
Confirmed! Thx!!
